db_etl_homepage_graphs/grapher.py

```diff
@@ -128,6 +174,50 @@ def get_vaccinations(date):
     return True
 
 
+def get_vaccinations_50_plus(date):
+    ts = datetime.strptime(date, "%Y-%m-%d")
+    partition = f"{ts:%Y_%-m_%-d}"
+
+    vax_query_50_plus = queries.VACCINATIONS_QUERY_50_PLUS.format(partition=partition)
+
+    session = Session()
+    conn = session.connection()
+    try:
+        resp50 = conn.execute(text(vax_query_50_plus))
+        values_50_plus = resp50.fetchall()
+    except Exception as err:
+        session.rollback()
+        raise err
+    finally:
+        session.close()
+
+    # This used for tracking most recent values
+    saved_data = {}
+
+    for item in values_50_plus:
+        # Checking if it's the newest data for the region
+        if (
+            item['area_type'] not in saved_data
+            or item['area_code'] not in saved_data[item['area_type']]
+            or item['date'] > saved_data[item['area_type']][item['area_code']]
+        ):
+            # Updating saved files dict
+            if item['area_type'] not in saved_data:
+                saved_data[item['area_type']] = {item['area_code']: item['date']}
+            elif item['area_code'] not in saved_data[item['area_type']]:
+                saved_data[item['area_type']][item['area_code']] = item['date']
+
+            store_data_50_plus(
```

Comment:
Should this have a better name?

--open question really but perhaps "upload_file" might be more understandable.